tests/main/econnreset/task.yaml

```diff
@@ -5,21 +5,25 @@ restore: |
 execute: |
     echo "Downloading a large snap in the background"
     su -c "/usr/bin/env SNAPD_DEBUG=1 snap download --edge test-snapd-huge 2>snap-download.log" test &
-    echo "Wait until it actually got some data for the snap"
-    while ! grep -q "Retrying.*download-snap/.*\.snap, attempt 1" snap-download.log; do
-        sleep 1
+
+    echo "Wait until the download started and downloaded more than 1 MB"
+    partial=test-snapd-huge_1.snap.partial
+    for i in $(seq 20); do
+        [ -f $partial ] && [ $(wc -c <"$partial") -gt 1048576 ] && break
+        sleep .5
     done
-    echo "Wait a little bit for the download to start, then force sending a RST package"
-    sleep 3
+    [ ! -f $partial ] || [ $(wc -c <"$partial") -eq 0 ] && exit 1
+
+    echo "Block the download using iptables"
     iptables -I OUTPUT -m owner --uid-owner $(id -u test) -j REJECT -p tcp --reject-with tcp-reset
+
     echo "Check that we retried"
     for i in $(seq 10); do
-        if grep -q "Retrying.*\.snap, attempt 2" snap-download.log; then
-            break
-        fi
+        [ grep -q "Retrying.*\.snap, attempt 2" snap-download.log ] && break
```

Comment:
(style nitpick): as above I would prefer the explicit if/then here (mostly for consistency with the rest of the shell code we have).